### PR TITLE
Skylanders: improve figure data

### PR DESCRIPTION
**Pull request by mandar1jn**
This aims to get figure data closer to what would be expected of a real figure. This way, figures generated using dolphin should, for example, be usable on magic NFC cards.

### CODE REVIEW

**Comment by mandar1jn** [comment]
congrats. Dolphin now has a very niche crc-64 to 48 algorithm no one will ever use outside of this.

File: Source/Core/Core/IOS/USB/Emulated/Skylander.cpp [hunk 1]
```diff
@@ -1223,13 +1262,18 @@ bool SkylanderPortal::CreateSkylander(const std::string& file_path, u16 m_sky_id
...
 
-  // SAK
+  // ATQA
+  file_data[6] = 0x01;
   file_data[7] = 0x0F;
 
+  // Year of production
+  // the last 2 digits of the year the tag was produced, interpreted as hexadecimal
+  // 2011 example:
+  // file_data[0xF] = 0x11;
```
Comment (mbc07) [inline Source/Core/Core/IOS/USB/Emulated/Skylander.cpp:1275]: Is that correct? 0x11 is 17 in decimal

Comment (mandar1jn) [reply Source/Core/Core/IOS/USB/Emulated/Skylander.cpp:1275]: "the last 2 digits of the year the tag was produced, interpreted as hexadecimal". With interpreted as hexadecimal, I mean 2011 becomes 0x11, 2012 becomes 0x12, 2013 becomes 0x13, etc. So yeah. That's correct. Wording suggestions are always appreciated

Comment (slx7R4GDZM) [reply Source/Core/Core/IOS/USB/Emulated/Skylander.cpp:1198]: >Wording suggestions are always appreciated

Seems like [binary-coded decimal](https://en.wikipedia.org/wiki/Binary-coded_decimal)?

File: Source/Core/Core/IOS/USB/Emulated/SkylanderCrypt.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  for (u32 i = 0; i < size; i++)
+  {
+    const u16 tmp = (crc >> 8) ^ buffer[i];
+    crc = (crc << 8) ^ CRC_CCITT_TABLE[tmp];
+  }
+
+  return crc;
+}
+
+// CRC-64 algorithm that is limited to 48 bits every itteration
```
Comment (windwakr) [inline Source/Core/Core/IOS/USB/Emulated/SkylanderCrypt.cpp:47]: iteration

**Comment by mandar1jn** [comment]
Waiting for #12090 to merge and then going to rewrite this from scratch

**Pull request closed without merging**